### PR TITLE
feat(shell-api, cli-repl): on-boot warnings for fake MongoDBs

### PR DESCRIPTION
**Pull request by himanshusinghs**
This PR adds an on-boot on invokable warning when connected to a fake mongodb deployment (DocumentDB for example).

Closes: MONGOSH-57

### CODE REVIEW

File: packages/shell-api/src/mongo.ts
Comment (himanshusinghs) [inline packages/shell-api/src/mongo.ts:430]: @mmarcon is reviewing the content and the link

**Comment by himanshusinghs** [comment]
Well I was looking at telemetry info for genuine check and realised that the work to determine wether a deployment is genuine or not is already done. I will probably make changes just to use that.

File: packages/shell-api/src/mongo.ts
Comment (himanshusinghs) [reply packages/shell-api/src/mongo.ts:451]: yea this is exactly what I stumbled upon while I was looking into telemetry for is_genuine, though there is a problem with the detection - our package `mongodb-build-info` expect `cmdLineOpts` to be of the form of legacy options but here in mongosh we throw when we get an error which is why the non-genuine mongodb's are never detected

Comment (himanshusinghs) [reply packages/shell-api/src/mongo.ts]: thank you anna! I addressed the changes in a recent commit

**Review by addaleax — APPROVED** [review]

**Comment by mmarcon** [comment]
If it's not too much trouble, I'd wait for the ACK from the docs team on the URL before merging. If we are not planning a release soon I am also happy to merge and change the URL in a separate PR.

File: packages/service-provider-server/src/cli-service-provider.ts
Comment (addaleax) [inline packages/service-provider-server/src/cli-service-provider.ts:244]: I’d make this a permalink since the line numbers might shift around on a given branch:

```suggestion
        // Ref: https://github.com/mongodb-js/mongodb-build-info/blob/9247eeba730a905397ad09fe5a377067edc49b34/index.js#L89
```

**Comment by addaleax** [comment]
@mmarcon Yeah, I’d like to do a release ideally this week, otherwise early next week … the docs link here is a dochub link, my understanding is that the idea here is that these serve as permanent link targets for external sources (like our products) with the intention that the actual location that they redirect to can be changed later as we need it, so we should be fine here?

@davemungo @ianf-mongodb maybe you can confirm?

File: packages/service-provider-server/src/cli-service-provider.ts
Comment (himanshusinghs) [reply packages/service-provider-server/src/cli-service-provider.ts:244]: oh that's right

**Comment by ianf-mongodb** [comment]
@mmarcon @addaleax that is correct currently the dochub entry is pointing at: https://www.mongodb.com/docs/manual/release-notes/4.2/#mongo-shell but can be updated if needed.

**Pull request merged**
